tools/osbuild-image-test

```diff
@@ -300,13 +300,14 @@ def inspect_image(test: TestCase, imi: ImageInfo, args: argparse.Namespace) -> b
 
     write_json(test.id, result, args.imi_output)
 
-    if not test.image_info:
-        print(f" {YELLOW}Warning{RESET}: No image info to compare to; ", end="")
-        return True
+    if not args.skip_inspection :
+        if not test.image_info:
+            print(f" {YELLOW}Warning{RESET}: No image info to compare to; ", end="")
+            return True
 
-    if result != test.image_info:
-        test.error = "image-info mismatch"
-        return False
+        if result != test.image_info:
+            test.error = "image-info mismatch"
+            return False
```

Comment:
If I'm reading this correctly, the `skip-inspection` flag only skips the comparison with the old image info, but the script will still read the old info, it just wont do anything with it?
Seems unnecessary and also hard to read.

This could use a rewrite to make things clearer.  If we don't need the old results and this flag "turns the tool
into an image-info generator", then it should just be that and not assume (or care) about any old image info.

Reply:
Yes, this is quite bad indeed. I need to rewrite this.